datafusion-federation/src/schema_cast/struct_cast.rs

```diff
@@ -0,0 +1,169 @@
+use arrow_json::ReaderBuilder;
+use datafusion::arrow::{
+    array::{Array, ArrayRef, StringArray},
+    datatypes::Field,
+    error::ArrowError,
+};
+use std::sync::Arc;
+
+pub type Result<T, E = crate::schema_cast::record_convert::Error> = std::result::Result<T, E>;
+
+pub(crate) fn cast_string_to_struct(
+    array: &dyn Array,
+    struct_field: Arc<Field>,
+) -> Result<ArrayRef, ArrowError> {
+    let string_array = array
+        .as_any()
+        .downcast_ref::<StringArray>()
+        .ok_or_else(|| ArrowError::CastError("Failed to downcast to StringArray".to_string()))?;
+
+    let mut decoder = ReaderBuilder::new_with_field(struct_field)
+        .build_decoder()
+        .map_err(|e| ArrowError::CastError(format!("Failed to create JSON decoder: {e}")))?;
+
+    for value in string_array {
+        match value {
+            Some(v) => {
+                decoder.decode(v.as_bytes()).map_err(|e| {
+                    ArrowError::CastError(format!("Failed to decode struct array: {e}"))
+                })?;
+            }
+            None => {
+                decoder.decode("null".as_bytes()).map_err(|e| {
```

Comment:
As we operate on decoder level and not on StructBuilder we pass "null" here to add Null entity. Unit tests include Null values.